pkg/virt-controller/services/template.go

```diff
@@ -1048,6 +1048,20 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 	selinuxType := t.clusterConfig.GetSELinuxLauncherType()
 	if selinuxType != virtconfig.DefaultSELinuxLauncherType {
 		pod.Spec.SecurityContext.SELinuxOptions = &k8sv1.SELinuxOptions{Type: selinuxType}
+		// By setting an SELinux option on the virt-launcher pod, we trigger this:
+		// https://github.com/kubernetes/kubernetes/issues/90759
+		// Since the compute container needs to be able to communicate with the rest of the pod,
+		//   we loop over all the containers and remove their SELinux categories.
+		for i := range pod.Spec.Containers {
+			if pod.Spec.Containers[i].Name != "compute" {
+				pod.Spec.Containers[i].SecurityContext = &k8sv1.SecurityContext{
```

Comment:
I think we should check if a container has a `SecurityContext` set already if it's set then add/replace SELinuxOptions.
For example, the readiness probe sets its own SecurityContext and I think we shouldn't override it.

Reply:
Good catch, thank you, fixed!